aws-crt/lib/aws-crt/io.rb

```diff
@@ -21,11 +21,17 @@ def initialize(max_threads = nil)
           @native = Aws::Crt.call do
             Aws::Crt::Native.event_loop_group_new(max_threads)
           end
+
+          ObjectSpace.define_finalizer(self, self.class.finalize(@native))
```

Comment:
written as specified in the [the-trouble-with-ruby-finalizers](https://www.mikeperham.com/2010/02/24/the-trouble-with-ruby-finalizers/)